From 0ed79841a3898fde8bbab673563ee672974c468b Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Fri, 18 Oct 2024 01:45:04 +0200
Subject: [PATCH] remove trailing whitespace

---
 dkms.8.in |  4 ++--
 dkms.in   | 16 ++++++++--------
 2 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/dkms.8.in b/dkms.8.in
index 5c829d9..634c697 100644
--- a/dkms.8.in
+++ b/dkms.8.in
@@ -333,8 +333,8 @@ combinations on the other kernel.
 .TP
 .B \-\-force
 This option can be used in conjunction with
-.B build, install 
-and 
+.B build, install
+and
 .B ldtarball
 to force copying over existing files.
 .TP
diff --git a/dkms.in b/dkms.in
index a8f2c6a..371290a 100644
--- a/dkms.in
+++ b/dkms.in
@@ -96,19 +96,19 @@ invoke_command()
     local cmd_mode="$4"
     local exitval=0
     local progresspid
-    
+
     [[ $verbose ]] && echo -e "$cmd" ||  echo -en "$cmd_description..."
 
     if [[ $cmd_mode == background && ! $verbose && $package_name != dkms*_test ]]; then
-            while true ; do 
+            while true ; do
                 sleep 3
                 echo  -n "."
             done &
             progresspid=$!
     fi
 
-    if [[ -n "$cmd_output_file" ]]; then 
-        ( eval "$cmd" ) >> "$cmd_output_file" 2>&1 
+    if [[ -n "$cmd_output_file" ]]; then
+        ( eval "$cmd" ) >> "$cmd_output_file" 2>&1
         exitval=$?
     elif [[ -z "$cmd_output_file" && $cmd_mode == background && ! $verbose ]]; then
         ( eval "$cmd" ) >/dev/null 2>&1
@@ -119,15 +119,15 @@ invoke_command()
     fi
 
     [ -n "$progresspid" ] && kill "$progresspid" >/dev/null 2>&1
-    
-    if (( exitval > 0)); then
+
+    if (( exitval > 0 )); then
         echo -en "(bad exit status: $exitval)"
         # Print the failing command without the clunky redirection
         [[ ! $verbose ]] && echo -en "\nFailed command:\n$1"
     else
         echo " done."
     fi
-    
+
     return "$exitval"
 }
 
@@ -1941,7 +1941,7 @@ make_tarball()
         die 6 "Failed to make tarball."
     fi
 
-    eval "$make_tarball_rm_temp_dir_name" 
+    eval "$make_tarball_rm_temp_dir_name"
     unset make_tarball_rm_temp_dir_name
 }
 
-- 
2.39.5

