From f56a77d7bcd06a968eb37f64bffc599b67f39766 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Fri, 18 Oct 2024 02:33:42 +0200
Subject: [PATCH] common.postinst: source /etc/dkms/framework.conf.d/*.conf,
 too

---
 dkms_common.postinst.in | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/dkms_common.postinst.in b/dkms_common.postinst.in
index 2c54058..e290028 100644
--- a/dkms_common.postinst.in
+++ b/dkms_common.postinst.in
@@ -114,9 +114,11 @@ if [ -f /etc/dkms/no-autoinstall ]; then
 fi
 
 # read framework configuration options
-if [ -r /etc/dkms/framework.conf ]; then
-    . /etc/dkms/framework.conf
-fi
+for fwcf in /etc/dkms/framework.conf /etc/dkms/framework.conf.d/*.conf ; do
+    if [ -f "$fwcf" ] && [ -r "$fwcf" ]; then
+        . "$fwcf"
+    fi
+done
 
 KERNELS=$(ls -dv @MODDIR@/*/build 2>/dev/null | cut -d/ -f4 || true)
 CURRENT_KERNEL=$(uname -r)
-- 
2.39.5

