Index: dkms/dkms_autoinstaller
===================================================================
--- dkms.orig/dkms_autoinstaller
+++ dkms/dkms_autoinstaller
@@ -47,7 +47,7 @@
 			kernel=$(uname -r)
 		fi
 		if [ -f /etc/dkms/no-autoinstall ]; then
-			log_daemon_msg "$prog: autoinstall for dkms modules has been disabled"
+			log_action_msg "$prog: autoinstall for dkms modules has been disabled"
 		else
 			log_daemon_msg "$prog: running auto installation service for kernel $kernel"
 			dkms autoinstall --kernelver $kernel
