Author: Andreas Beckmann <anbe@debian.org>
Description: spelling corrections

Index: dkms/dkms_common.postinst
===================================================================
--- dkms.orig/dkms_common.postinst
+++ dkms/dkms_common.postinst
@@ -254,7 +254,7 @@
             dkms_status=$(dkms status -m $NAME -v $VERSION -k $KERNEL $ARCH)
         else
             echo "Module build for kernel $KERNEL was skipped since the"
-            echo "kernel headers for this kernel does not seem to be installed."
+            echo "kernel headers for this kernel do not seem to be installed."
         fi
     fi
 
